druid/examples/image.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This example shows how to draw an png image.
+//!
+//! Requires the non-default "image" feature to be enabled:
+//! `cargo run --example image --features "images"`
+//!
+
+#[cfg(not(feature = "image"))]
+fn main() {
+    eprintln!("This examples requires the \"image\" feature to be enabled:");
+    eprintln!("cargo run --example image --features \"image\"");
+}
+
+#[cfg(feature = "image")]
+fn main() {
+    use std::str::FromStr;
+
+    use druid::{
+        widget::{FillStrat, Flex, Image, ImageData, WidgetExt},
+        AppLauncher, Widget, WindowDesc,
+    };
+
+    fn ui_builder() -> impl Widget<u32> {
+        let png_data = ImageData::from_str("examples/pngexample.png").unwrap();
+
+        let mut col = Flex::column();
+
+        col.add_child(Image::new(png_data.clone()).fix_width(100.0).center(), 1.0);
+
+        /* 
+        // If you want to change the fill stratagy you can but you need the widget to be mut
+        let mut otherimage = Image::new(png_data);
+        otherimage.set_fill(FillStrat::FitWidth);
+        */
+
+        let otherimage = Image::new(png_data);
+        let otherimage = otherimage.fill_mode(FillStrat::FitWidth);
```

Comment:
Is this the way to do it?

re-letting seems weird but i haven't got my head round this rust patten, its such a bad idea in some langues and i half get why its better for rust but im get this wrong sometimes.

Reply:
nope, it should look like,

```rust
let my_image = Image::new(png_data).fill_mode(FillStart::FitWidth);
```

You can take a look at widgets like `Container` for other examples of this pattern.

Reply:
thanks